app/controllers/declaration/InlandTransportDetailsController.scala

```diff
@@ -68,18 +68,6 @@ class InlandTransportDetailsController @Inject()(
       )
   }
 
-  private def updateCache(formData: WarehouseDetails)(implicit request: JourneyRequest[AnyContent]): Future[Option[ExportsDeclaration]] =
-    updateExportsDeclarationSyncDirect { model =>
-      val warehouseDetails = model.locations.warehouseIdentification
-        .map(
-          dbWarehouseDetails =>
-            WarehouseDetails(
-              identificationNumber = dbWarehouseDetails.identificationNumber,
-              supervisingCustomsOffice = dbWarehouseDetails.supervisingCustomsOffice,
-              inlandModeOfTransportCode = formData.inlandModeOfTransportCode
-          )
-        )
-        .getOrElse(WarehouseDetails(inlandModeOfTransportCode = formData.inlandModeOfTransportCode))
-      model.copy(locations = model.locations.copy(warehouseIdentification = Some(warehouseDetails)))
-    }
+  private def updateCache(formData: InlandModeOfTransportCode)(implicit request: JourneyRequest[AnyContent]): Future[Option[ExportsDeclaration]] =
+    updateExportsDeclarationSyncDirect(model => model.copy(locations = model.locations.copy(inlandModeOfTransportCode = Some(formData))))
```

Comment:
please move update logic to method on model

Reply:
Do we have examples of this being done in our project?

Reply:
Inside `ExportsDeclaration` model you can find it